### PR TITLE
[API Compatiblity]  paddle.roll、paddle.flatten and paddle.Tensor.flatten sink into C++

### PR DESCRIPTION
**Pull request by zhengshengning**
### PR Category

Operator Mechanism

### PR Types

New features

### Description

下沉paddle.roll、paddle.flatten 和 paddle.Tensor.flatten 到C++层，支持Torch的传参方法。
**流程：**

1. 在ops.yaml增加 python_api 配置，增加映射关系和PreProcess函数；
2. 在_paddle_docs.py增加英文文档信息，add_doc_and_signature绑定；
3. 删掉原有实现，使用from _C_ops import xxx 代替;
4. 在arg_pre_process.h与arg_pre_process.cpp中实现PreProcess函数；
5. 删掉报错的老静态图（Old IR）的单测；
6. 增加单测

**其他：**
修复python_c_gen代码生成的bug。

避免nn.functional.sigmoid 引起的单测报错

pcard-67164

### CODE REVIEW

**Review by zhwesky2010 — COMMENTED** [review]
LGTM for delete UT

**Review by zhwesky2010 — DISMISSED** [review]

**Review by XieYunshen — DISMISSED** [review]

**Review by DanielSun11 — DISMISSED** [review]
LGTM

File: paddle/fluid/pybind/arg_pre_process.cc [hunk 1]
```diff
@@ -19,11 +19,50 @@
...
+          true,
+          common::errors::InvalidArgument("axis is out of range, it should be "
+                                          "in range [%d, %d), but received %ld",
+                                          -len_origin_shape,
+                                          len_origin_shape,
+                                          (*axis)[i]));
+    }
+  } else {
+    axis = new IntVector();
+  }
```
Comment (zyfncg) [inline paddle/fluid/pybind/arg_pre_process.cc:45]: 这个环节处理的作用什么？看起来只有检查

Comment (zhengshengning) [reply paddle/fluid/pybind/arg_pre_process.cc:45]: 原来python端在_C_ops前会检查参数是否合法，现在下沉以后，同样加个参数检查

Comment (zyfncg) [reply paddle/fluid/pybind/arg_pre_process.cc:45]: 如果是某些参数值的有效性检查，可以放到infermeta函数中

Comment (wanghuancoder) [reply paddle/fluid/pybind/arg_pre_process.cc:45]: 云飞，C++下沉的时候没有考虑到这一点。确实有部分检查可以放到InferMeta中。有些可能不可以。因为到InferMeta就丢失信息了。或者将来可以都把有效性检查放到C_ops这层。更合理。这个可以有机会、有人力的时候统一梳理。目前CI非常紧张，能走排队到这步不容易。周五deadline。我们先合入吧~

Comment (zyfncg) [reply paddle/fluid/pybind/arg_pre_process.cc:45]: 这里可以再重新设计下

**Review by zyfncg — DISMISSED** [review]

**Review by wanghuancoder — APPROVED** [review]
